### ISSUE TITLE
node-v12.13.0 error

### ISSUE THREAD

**Issue by loretoparisi** [Wulf/nodehun#74]
I get this compilation error while using `node lts`, hence ` node-v12.13.0`
```
> nodehun@2.0.12 install /Users/loretoparisi/Documents/Projects/musixmatch-ai-console/webservice/node_modules/nodehun
> node-gyp rebuild

  CXX(target) Release/obj.target/hunspell/src/hunspell/src/hunspell/affentry.o
  CXX(target) Release/obj.target/hunspell/src/hunspell/src/hunspell/affixmgr.o
  CXX(target) Release/obj.target/hunspell/src/hunspell/src/hunspell/csutil.o
  CXX(target) Release/obj.target/hunspell/src/hunspell/src/hunspell/filemgr.o
  CXX(target) Release/obj.target/hunspell/src/hunspell/src/hunspell/hashmgr.o
  CXX(target) Release/obj.target/hunspell/src/hunspell/src/hunspell/hunspell.o
  CXX(target) Release/obj.target/hunspell/src/hunspell/src/hunspell/hunzip.o
  CXX(target) Release/obj.target/hunspell/src/hunspell/src/hunspell/phonet.o
  CXX(target) Release/obj.target/hunspell/src/hunspell/src/hunspell/replist.o
  CXX(target) Release/obj.target/hunspell/src/hunspell/src/hunspell/strmgr.o
../src/hunspell/src/hunspell/strmgr.cxx:79:19: warning: format string is not a string literal (potentially insecure) [-Wformat-security]
  fprintf(stderr, err);
                  ^~~
../src/hunspell/src/hunspell/strmgr.cxx:79:19: note: treat the string as an argument to avoid this
  fprintf(stderr, err);
                  ^
                  "%s", 
1 warning generated.
  CXX(target) Release/obj.target/hunspell/src/hunspell/src/hunspell/suggestmgr.o
  LIBTOOL-STATIC Release/hunspell.a
  CXX(target) Release/obj.target/nodehun/src/nodehun.o
In file included from ../src/nodehun.cpp:7:
In file included from ../src/post0.12.0/nodehun.cpp:1:
../src/post0.12.0/nodehun.hpp:131:24: error: no template named 'Handle' in namespace 'v8'
  static void Init(v8::Handle<v8::Object>, v8::Handle<v8::Object>);
                   ~~~~^
../src/post0.12.0/nodehun.hpp:131:48: error: no template named 'Handle' in namespace 'v8'
  static void Init(v8::Handle<v8::Object>, v8::Handle<v8::Object>);
                                           ~~~~^
In file included from ../src/nodehun.cpp:7:
../src/post0.12.0/nodehun.cpp:7:32: error: variable has incomplete type 'void'
void Nodehun::SpellDictionary::Init(Handle<Object> exports, Handle<Object> module)
                               ^
../src/post0.12.0/nodehun.cpp:7:37: error: use of undeclared identifier 'Handle'
void Nodehun::SpellDictionary::Init(Handle<Object> exports, Handle<Object> module)
                                    ^
../src/post0.12.0/nodehun.cpp:7:44: error: 'Object' does not refer to a value
void Nodehun::SpellDictionary::Init(Handle<Object> exports, Handle<Object> module)
                                           ^
/Users/loretoparisi/Library/Caches/node-gyp/12.13.0/include/node/v8.h:3400:17: note: declared here
class V8_EXPORT Object : public Value {
                ^
In file included from ../src/nodehun.cpp:7:
../src/post0.12.0/nodehun.cpp:7:52: error: use of undeclared identifier 'exports'
void Nodehun::SpellDictionary::Init(Handle<Object> exports, Handle<Object> module)
                                                   ^
../src/post0.12.0/nodehun.cpp:7:61: error: use of undeclared identifier 'Handle'
void Nodehun::SpellDictionary::Init(Handle<Object> exports, Handle<Object> module)
                                                            ^
../src/post0.12.0/nodehun.cpp:7:68: error: 'Object' does not refer to a value
void Nodehun::SpellDictionary::Init(Handle<Object> exports, Handle<Object> module)
                                                                   ^
/Users/loretoparisi/Library/Caches/node-gyp/12.13.0/include/node/v8.h:3400:17: note: declared here
class V8_EXPORT Object : public Value {
                ^
In file included from ../src/nodehun.cpp:7:
../src/post0.12.0/nodehun.cpp:7:76: error: use of undeclared identifier 'module'
void Nodehun::SpellDictionary::Init(Handle<Object> exports, Handle<Object> module)
                                                                           ^
../src/post0.12.0/nodehun.cpp:7:83: error: expected ';' after top level declarator
void Nodehun::SpellDictionary::Init(Handle<Object> exports, Handle<Object> module)
                                                                                  ^
                                                                                  ;
10 errors generated.
make: *** [Release/obj.target/nodehun/src/nodehun.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:210:5)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 18.6.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/loretoparisi/Documents/Projects/node_modules/nodehun
gyp ERR! node -v v12.13.0
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok 
npm WARN musixmatch-ai-console@5.6.7-20191022 No repository field.
npm WARN musixmatch-ai-console@5.6.7-20191022 No license field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nodehun@2.0.12 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the nodehun@2.0.12 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/loretoparisi/.npm/_logs/2019-10-24T09_49_25_397Z-debug.log
```

**Comment by Wulf** [comment]
Hey @loretoparisi, thanks for your patience with this issue. Could I request you try installing nodehun 3? The API has slightly changed, you can find the documentation [here](https://github.com/Wulf/nodehun/blob/n-api/src/Nodehun.d.ts).

```
npm i -S wulf/nodehun#n-api
```

**Comment by loretoparisi** [comment]
@Wulf ok thank you I will try it.

**Comment by loretoparisi** [comment]
@Wulf I get a `error Cannot read property 'resolve' of undefined` error when trying to install to 

```
$ node --version
v12.13.0
$ npm --version
6.12.0
``` 

```
19 silly saveTree └── nodehun@3.0.0
20 verbose stack TypeError: Cannot read property 'resolve' of undefined
20 verbose stack     at regFetch (/usr/local/lib/node_modules/npm/node_modules/npm-registry-fetch/index.js:76:23)
20 verbose stack     at fetchPackument (/usr/local/lib/node_modules/npm/node_modules/pacote/lib/fetchers/registry/packument.js:42:10)
20 verbose stack     at packument (/usr/local/lib/node_modules/npm/node_modules/pacote/lib/fetchers/registry/packument.js:20:10)
20 verbose stack     at getManifest (/usr/local/lib/node_modules/npm/node_modules/pacote/lib/fetchers/registry/manifest.js:22:10)
20 verbose stack     at manifest (/usr/local/lib/node_modules/npm/node_modules/pacote/lib/fetchers/registry/manifest.js:13:10)
20 verbose stack     at Object.manifest (/usr/local/lib/node_modules/npm/node_modules/pacote/lib/fetchers/registry/index.js:17:12)
20 verbose stack     at Object.Fetcher#manifest [as manifest] (/usr/local/lib/node_modules/npm/node_modules/genfun/lib/genfun.js:15:38)
20 verbose stack     at manifest (/usr/local/lib/node_modules/npm/node_modules/pacote/lib/fetch.js:23:18)
20 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/pacote/manifest.js:24:12
20 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/promise-inflight/inflight.js:29:24
20 verbose stack     at Promise._execute (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/debuggability.js:313:9)
20 verbose stack     at Promise._resolveFromExecutor (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:488:18)
20 verbose stack     at new Promise (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:79:10)
20 verbose stack     at _inflight (/usr/local/lib/node_modules/npm/node_modules/promise-inflight/inflight.js:28:25)
20 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/promise-inflight/inflight.js:22:14
20 verbose stack     at tryCatcher (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
22 verbose Darwin 18.6.0
23 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "i" "-S" "wulf/nodehun#n-api"
24 verbose node v12.13.0
25 verbose npm  v6.12.0
26 error Cannot read property 'resolve' of undefined
27 verbose exit [ 1, true ]
```

**Comment by Wulf** [comment]
@loretoparisi, thanks for trying this! I'm sad to see the install command fail -- do you think it's possibly something to do with `npm` itself? [This](https://github.com/nodejs/help/issues/1941) is all I found about the error. Kindly try updating to the latest npm: 

```
npm i -g npm@latest
```

**Comment by loretoparisi** [comment]
@Wulf so This is what I did now:

```
$ npm --version
6.13.0
$ rm -rf node_modules/
$ npm i -S wulf/nodehun#n-api

> nodehun@3.0.0 install /Users/loretoparisi/Documents/Projects/node-sdk/node_modules/nodehun
> node-gyp rebuild

  CC(target) Release/obj.target/nothing/../node-addon-api/src/nothing.o
  LIBTOOL-STATIC Release/nothing.a
  CXX(target) Release/obj.target/hunspell/hunspell/src/hunspell/affentry.o
../hunspell/src/hunspell/affentry.cxx:544:47: warning: while loop has empty body [-Wempty-body]
                        while (p && *p != ']' && (p = nextchar(p)));
                                                                   ^
../hunspell/src/hunspell/affentry.cxx:544:47: note: put the semicolon on a separate line to silence this warning
1 warning generated.
  CXX(target) Release/obj.target/hunspell/hunspell/src/hunspell/affixmgr.o
In file included from ../hunspell/src/hunspell/affixmgr.cxx:13:
../hunspell/src/hunspell/affentry.hxx:30:105: warning: implicit conversion of NULL constant to 'unsigned short' [-Wnull-conversion]
  struct hentry *      check_twosfx(const char * word, int len, char in_compound, const FLAG needflag = NULL);
                                                                                                      ~ ^~~~
                                                                                                        0
../hunspell/src/hunspell/affentry.hxx:93:114: warning: implicit conversion of NULL constant to 'unsigned short' [-Wnull-conversion]
  struct hentry *   check_twosfx(const char * word, int len, int optflags, PfxEntry* ppfx, const FLAG needflag = NULL);
                                                                                                               ~ ^~~~
                                                                                                                 0
../hunspell/src/hunspell/affixmgr.cxx:3671:65: warning: 'strncmp' call operates on objects of type 'const char' while the size is based on a different type
      'const char *' [-Wsizeof-pointer-memaccess]
                             if (strncmp(piece, keyword, sizeof(keyword)) != 0) {
                                                ~~~~~~~         ^~~~~~~
../hunspell/src/hunspell/affixmgr.cxx:3671:65: note: did you mean to provide an explicit length?
                             if (strncmp(piece, keyword, sizeof(keyword)) != 0) {
                                                                ^~~~~~~
3 warnings generated.
  CXX(target) Release/obj.target/hunspell/hunspell/src/hunspell/csutil.o
  CXX(target) Release/obj.target/hunspell/hunspell/src/hunspell/dictmgr.o
  CXX(target) Release/obj.target/hunspell/hunspell/src/hunspell/filemgr.o
  CXX(target) Release/obj.target/hunspell/hunspell/src/hunspell/hashmgr.o
In file included from ../hunspell/src/hunspell/hashmgr.cxx:9:
../hunspell/src/hunspell/hashmgr.hxx:17:21: warning: private field 'userword' is not used [-Wunused-private-field]
  int               userword;
                    ^
1 warning generated.
  CXX(target) Release/obj.target/hunspell/hunspell/src/hunspell/hunspell.o
  CXX(target) Release/obj.target/hunspell/hunspell/src/hunspell/hunzip.o
  CXX(target) Release/obj.target/hunspell/hunspell/src/hunspell/phonet.o
  CXX(target) Release/obj.target/hunspell/hunspell/src/hunspell/replist.o
  CXX(target) Release/obj.target/hunspell/hunspell/src/hunspell/strmgr.o
  CXX(target) Release/obj.target/hunspell/hunspell/src/hunspell/suggestmgr.o
  LIBTOOL-STATIC Release/hunspell.a
  CXX(target) Release/obj.target/Nodehun/src/index.o
In file included from ../src/index.cc:2:
In file included from ../src/Nodehun.cc:1:
In file included from ../src/Nodehun.h:5:
../src/HunspellContext.h:43:8: error: 'shared_mutex' is unavailable: introduced in macOS 10.12
  std::shared_mutex rwLock;
       ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/shared_mutex:180:58: note: 'shared_mutex' has been explicitly
      marked unavailable here
class _LIBCPP_TYPE_VIS _LIBCPP_AVAILABILITY_SHARED_MUTEX shared_mutex
                                                         ^
In file included from ../src/index.cc:2:
In file included from ../src/Nodehun.cc:4:
In file included from ../src/Async/AddDictionaryWorker.cc:3:
../src/Async/Worker.cc:11:165: warning: field 'deferred' will be initialized after field 'context' [-Wreorder]
  ...context, Napi::Promise::Deferred const &d, const char* resource_name) : AsyncWorker(get_fake_callback(d.Env()).Value(), resource_name), deferred(d), context(co...
                                                                                                                                             ^
../src/Async/Worker.cc:12:123: warning: field 'deferred' will be initialized after field 'context' [-Wreorder]
    Worker(HunspellContext* context, Napi::Promise::Deferred const &d) : AsyncWorker(get_fake_callback(d.Env()).Value()), deferred(d), context(context) {}
                                                                                                                          ^
2 warnings and 1 error generated.
make: *** [Release/obj.target/Nodehun/src/index.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (events.js:210:5)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 18.6.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/loretoparisi/Documents/Projects/node-sdk/node_modules/nodehun
gyp ERR! node -v v12.13.0
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok 
npm WARN musixmatch-intelligence-node-sdk@3.0.2 No license field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! nodehun@3.0.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the nodehun@3.0.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/loretoparisi/.npm/_logs/2019-11-16T12_26_06_538Z-debug.log
```

**Comment by loretoparisi** [comment]
More info from logging file

```
 info it worked if it ends with ok
1 verbose cli [
1 verbose cli   '/usr/local/bin/node',
1 verbose cli   '/usr/local/bin/npm',
1 verbose cli   'i',
1 verbose cli   '-S',
1 verbose cli   'wulf/nodehun#n-api'
1 verbose cli ]
2 info using npm@6.13.0
3 info using node@v12.13.0
4 verbose npm-session 3e5883bfc4f5fb13
5 silly install loadCurrentTree
6 silly install readLocalPackageData
7 silly pacote git manifest for undefined@github:wulf/nodehun#n-api fetched in 1609ms
8 timing stage:loadCurrentTree Completed in 1637ms
9 silly install loadIdealTree
10 silly install cloneCurrentTreeToIdealTree
11 timing stage:loadIdealTree:cloneCurrentTree Completed in 0ms
12 silly install loadShrinkwrap
13 timing stage:loadIdealTree:loadShrinkwrap Completed in 2ms
14 silly install loadAllDepsIntoIdealTree
15 silly resolveWithNewModule nodehun@3.0.0 checking installable status
16 http fetch GET 304 https://registry.npmjs.org/node-addon-api 398ms (from cache)
17 silly pacote range manifest for node-addon-api@* fetched in 404ms
18 silly resolveWithNewModule node-addon-api@1.7.1 checking installable status
19 timing stage:loadIdealTree:loadAllDepsIntoIdealTree Completed in 411ms
20 timing stage:loadIdealTree Completed in 417ms
21 silly currentTree node-sdk@3.0.2
22 silly idealTree node-sdk@3.0.2
22 silly idealTree ├── node-addon-api@1.7.1
22 silly idealTree └── nodehun@3.0.0
23 silly install generateActionsToTake
24 timing stage:generateActionsToTake Completed in 6ms
25 silly diffTrees action count 2
26 silly diffTrees add node-addon-api@1.7.1
27 silly diffTrees add nodehun@3.0.0
28 silly decomposeActions action count 16
29 silly decomposeActions fetch node-addon-api@1.7.1
30 silly decomposeActions extract node-addon-api@1.7.1
31 silly decomposeActions preinstall node-addon-api@1.7.1
32 silly decomposeActions build node-addon-api@1.7.1
33 silly decomposeActions install node-addon-api@1.7.1
34 silly decomposeActions postinstall node-addon-api@1.7.1
35 silly decomposeActions finalize node-addon-api@1.7.1
36 silly decomposeActions refresh-package-json node-addon-api@1.7.1
37 silly decomposeActions fetch nodehun@3.0.0
38 silly decomposeActions extract nodehun@3.0.0
39 silly decomposeActions preinstall nodehun@3.0.0
40 silly decomposeActions build nodehun@3.0.0
41 silly decomposeActions install nodehun@3.0.0
42 silly decomposeActions postinstall nodehun@3.0.0
43 silly decomposeActions finalize nodehun@3.0.0
44 silly decomposeActions refresh-package-json nodehun@3.0.0
45 silly install executeActions
46 silly doSerial global-install 16
47 verbose correctMkdir /Users/loretoparisi/.npm/_locks correctMkdir not in flight; initializing
48 verbose lock using /Users/loretoparisi/.npm/_locks/staging-965a37ac906df377.lock for /Users/loretoparisi/Documents/Projects/node-sdk/node_modules/.staging
49 silly doParallel extract 2
50 silly extract node-addon-api@1.7.1
51 silly extract nodehun@3.0.0
52 silly tarball trying node-addon-api@* by hash: sha512-2+DuKodWvwRTrCfKOeR24KIc5unKjOh8mz17NCzVnHWfjAdDqbfbjqh7gUT+BkXBRQM52+xCHciKWonJ3CbJMQ==
53 silly tarball no local data for github:wulf/nodehun#n-api. Extracting by manifest.
54 silly extract node-addon-api@* extracted to /Users/loretoparisi/Documents/Projects/node-sdk/node_modules/.staging/node-addon-api-9799f9ec (106ms)
55 silly extract github:wulf/nodehun#n-api extracted to /Users/loretoparisi/Documents/Projects/node-sdk/node_modules/.staging/nodehun-e89a6d61 (236ms)
56 timing action:extract Completed in 238ms
57 silly doReverseSerial unbuild 16
58 silly doSerial remove 16
59 silly doSerial move 16
60 silly doSerial finalize 16
61 silly finalize /Users/loretoparisi/Documents/Projects/node-sdk/node_modules/node-addon-api
62 silly finalize /Users/loretoparisi/Documents/Projects/node-sdk/node_modules/nodehun
63 timing action:finalize Completed in 7ms
64 silly doParallel refresh-package-json 2
65 silly refresh-package-json /Users/loretoparisi/Documents/Projects/node-sdk/node_modules/node-addon-api
66 silly refresh-package-json /Users/loretoparisi/Documents/Projects/node-sdk/node_modules/nodehun
67 timing action:refresh-package-json Completed in 11ms
68 silly doParallel preinstall 2
69 silly preinstall node-addon-api@1.7.1
70 info lifecycle node-addon-api@1.7.1~preinstall: node-addon-api@1.7.1
71 silly preinstall nodehun@3.0.0
72 info lifecycle nodehun@3.0.0~preinstall: nodehun@3.0.0
73 timing action:preinstall Completed in 1ms
74 silly doSerial build 16
75 silly build node-addon-api@1.7.1
76 info linkStuff node-addon-api@1.7.1
77 silly linkStuff node-addon-api@1.7.1 has /Users/loretoparisi/Documents/Projects/node-sdk/node_modules as its parent node_modules
78 silly build nodehun@3.0.0
79 info linkStuff nodehun@3.0.0
80 silly linkStuff nodehun@3.0.0 has /Users/loretoparisi/Documents/Projects/node-sdk/node_modules as its parent node_modules
81 timing action:build Completed in 1ms
82 silly doSerial global-link 16
83 silly doParallel update-linked 0
84 silly doSerial install 16
85 silly install node-addon-api@1.7.1
86 info lifecycle node-addon-api@1.7.1~install: node-addon-api@1.7.1
87 silly install nodehun@3.0.0
88 info lifecycle nodehun@3.0.0~install: nodehun@3.0.0
89 verbose lifecycle nodehun@3.0.0~install: unsafe-perm in lifecycle true
90 verbose lifecycle nodehun@3.0.0~install: PATH: /usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/Users/loretoparisi/Documents/Projects/node-sdk/node_modules/nodehun/node_modules/.bin:/Users/loretoparisi/Documents/Projects/node-sdk/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/share/dotnet:~/.dotnet/tools:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Wireshark.app/Contents/MacOS:/Users/loretoparisi/Documents/Projects/ImageMagick-7.0.7/bin:/Users/loretoparisi/torch/install/bin:/Users/loretoparisi/apache-maven-3.3.9/bin:/usr/local/opt/gettext/bin
91 verbose lifecycle nodehun@3.0.0~install: CWD: /Users/loretoparisi/Documents/Projects/node-sdk/node_modules/nodehun
92 silly lifecycle nodehun@3.0.0~install: Args: [ '-c', 'node-gyp rebuild' ]
93 timing audit submit Completed in 866ms
94 http fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/audits/quick 865ms
95 timing audit body Completed in 0ms
96 silly lifecycle nodehun@3.0.0~install: Returned: code: 1  signal: null
97 info lifecycle nodehun@3.0.0~install: Failed to exec install script
98 timing action:install Completed in 7304ms
99 verbose unlock done using /Users/loretoparisi/.npm/_locks/staging-965a37ac906df377.lock for /Users/loretoparisi/Documents/Projects/node-sdk/node_modules/.staging
100 timing stage:rollbackFailedOptional Completed in 45ms
101 timing stage:runTopLevelLifecycles Completed in 9687ms
102 silly saveTree node-sdk@3.0.2
102 silly saveTree └─┬ nodehun@3.0.0
102 silly saveTree   └── node-addon-api@1.7.1
103 warn node-sdk@3.0.2 No license field.
104 verbose stack Error: nodehun@3.0.0 install: `node-gyp rebuild`
104 verbose stack Exit status 1
104 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
104 verbose stack     at EventEmitter.emit (events.js:210:5)
104 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
104 verbose stack     at ChildProcess.emit (events.js:210:5)
104 verbose stack     at maybeClose (internal/child_process.js:1021:16)
104 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
105 verbose pkgid nodehun@3.0.0
106 verbose cwd /Users/loretoparisi/Documents/node-sdk
107 verbose Darwin 18.6.0
108 verbose argv "/usr/local/bin/node" "/usr/local/bin/npm" "i" "-S" "wulf/nodehun#n-api"
109 verbose node v12.13.0
110 verbose npm  v6.13.0
111 error code ELIFECYCLE
112 error errno 1
113 error nodehun@3.0.0 install: `node-gyp rebuild`
113 error Exit status 1
114 error Failed at the nodehun@3.0.0 install script.
114 error This is probably not a problem with npm. There is likely additional logging output above.
115 verbose exit [ 1, true ]

```

**Comment by Wulf** [comment]
@loretoparisi thanks again for your time.

Could you please try installing it once more (`npm i -S wulf/nodehun#n-api`). I've lifted the C++17 requirement by switching from `std::shared_mutex` to libuv's `uv_rwlock_t` since it ships with Node.js by default.

**Comment by loretoparisi** [comment]
@Wulf okay let me try today again.

**Comment by loretoparisi** [comment]
@Wulf I get
```
$ npm i -S wulf/nodehun#n-api
npm ERR! Cannot read property 'resolve' of undefined
```

**Comment by Wulf** [comment]
It seems npm 6.12.0 is broken -- we got the same error earlier. Could you update to the latest again?

 `npm i -g npm@latest`

It worked when you used 6.13.0.
Would you prefer if I published the package instead?

**Comment by loretoparisi** [comment]
@Wulf uhm I think it would be safer to double check this, in my case (and I think several other) it may result in a package disruption :D let me try again thanks!

**Comment by Wulf** [comment]
Cool, sounds good, thanks a bunch :)

**Comment by loretoparisi** [comment]
@Wulf Hello, so I have tested

```json
{
"nodehun": "github:wulf/nodehun#n-api",
}
```

against both node v11 and v12 for back compatibility:

```
$ npm --version
6.13.1
$ node --version
v11.1.0
```

and

```
$ node --version
v12.13.0
$ npm --version
6.13.1
```

so for me it's ok to move to latest 🚀 

Sidenote: a lot of packages breaks with node v12, among then `node-svm`, `node-java`, ...so I will stuck with node v11 until I get rid of the other ones...

**Comment by loretoparisi** [comment]
@Wulf I'm having this new issue `npm ERR! undefined ls-remote -h -t ssh://git@github.com/wulf/nodehun.git`
I'm using the new`"nodehun": "github:wulf/nodehun#n-api"`

**Comment by Wulf** [comment]
> Sidenote: a lot of packages breaks with node v12, among then nan by nodejs (!)...I will be stuck to node v11 for a while, but you can safely update nodehun at least!

Yikes, I didn't know `nan` was also broken by v12! Anyway, thanks for testing this Loreto :)

With regards to the `npm ERR! undefined ls-remote...` -- I'm going to assume that some npm versions just don't like installing from a specific branch. I'll test out `"nodehun": "^3.0.0"` after publishing. Hopefully that will resolve the issue.

**Comment by Wulf** [comment]
Hey @loretoparisi after publishing and installing 3.0.0, everything worked just fine. I found and fixed an error with the type declaration file (will release the fix in 3.0.1).

Let me know if `"nodehun": "^3.0.0"` fixes the `npm ERR! undefined ls-remote...` error.
If not, it might be the lock file (`package-lock.json` or `yarn.lock`). Try deleting that and running `npm install` again.

Thanks again for your time my dude!

**Comment by loretoparisi** [comment]
@Wulf confirmed! It works fine now with 

```
$ node --version
v12.13.1
$ npm --version
6.12.1
```
